docs/conf.py

```diff
@@ -197,7 +198,7 @@ def run(self):
 latex_logo = str(CURR_PATH / 'logo' / 'LightGBM_logo_black_text_small.png')
 
 
-def generate_doxygen_xml(app):
+def generate_doxygen_xml(app: Any) -> None:
```

Comment:
Maybe use `sphinx.application.Sphinx` for `app` (here and at L246 and L308)?
https://www.sphinx-doc.org/en/master/extdev/appapi.html

Reply:
Sure, can do. I was just following the documentation below.

```text
    Parameters
    ----------
    app : object
        The application object representing the Sphinx process.
```

Reply:
I think we can update docstring as well.

Reply:
sure! Updated in https://github.com/microsoft/LightGBM/pull/4526/commits/81f0a8259078df7abe12f92b0bcee811129db127